src/shogun/labels/BinaryLabelEncoder.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Yuhui Liu
+ *
+ */
+#ifndef _BINARYLABELENCODER__H__
+#define _BINARYLABELENCODER__H__
+
+#include <memory>
+#include <shogun/base/SGObject.h>
+#include <shogun/labels/BinaryLabels.h>
+#include <shogun/labels/DenseLabels.h>
+#include <shogun/labels/LabelEncoder.h>
+#include <shogun/lib/SGVector.h>
+
+namespace shogun
+{
+
+	class BinaryLabelEncoder : public LabelEncoder
+	{
+	public:
+		BinaryLabelEncoder() = default;
+
+		~BinaryLabelEncoder() = default;
+
+		/** Fit label encoder
+		 *
+		 * @param Target values.
+		 * @return SGVector which contains unique labels.
+		 */
+		SGVector<float64_t> fit(const std::shared_ptr<Labels>& labs) override
+		{
+			const auto result_vector = labs->as<DenseLabels>()->get_labels();
+			fit_impl(result_vector);
+			require(
+			    unique_labels.size() == 2,
+			    "BinaryLabel should contain only two elements");
+
+			return SGVector<float64_t>(
+			    unique_labels.begin(), unique_labels.end());
+		}
+		/** Transform labels to normalized encoding.
+		 *
+		 * @param  Target values to be transformed.
+		 * @return Labels transformed to be normalized.
+		 */
+		std::shared_ptr<Labels>
+		transform(const std::shared_ptr<Labels>& labs) override
+		{
+			const auto result_vector = labs->as<DenseLabels>()->get_labels();
+			require(
+			    std::set<float64_t>(result_vector.begin(), result_vector.end())
+			            .size() == 2,
+			    "BinaryLabel should contain only two elements");
+			auto transformed_vec = transform_impl(result_vector);
+
+			std::transform(
+			    transformed_vec.begin(), transformed_vec.end(),
+			    transformed_vec.begin(), [](float64_t e) {
+				    if (std::abs(e - 0.0) <=
+				        std::numeric_limits<float64_t>::epsilon())
+					    return -1.0;
+				    else
+					    return e;
```

Comment:
Here you assume that the labels are either 0 or 1 right?

Reply:
The labels have been transformed by `auto transformed_vec = transform_impl(result_vector);`, so the labels should only contain 0 or 1 now.

Reply:
Oh I see, so if you have something like {0,10} -> {0,1} -> {-1,+1}? Have you checked everything works fine with negative labels as well?

Reply:
yes, it exactly is what I did, for the situation with negative labels, i will add more test.